serde_codegen_internals/src/attr.rs

```diff
@@ -746,6 +785,64 @@ fn get_ser_and_de<T, F>(cx: &Ctxt,
     Ok((ser_item.get(), de_item.get()))
 }
 
+type TypeConversions<T> = (Option<T>, Option<T>, Option<T>, Option<T>);
+
+fn get_from_into_as(cx: &Ctxt,
```

Comment:
Compile-fail tests are supposed to test things that fail to compile. For example [this test](https://github.com/serde-rs/serde/blob/c488cec6412202280b99748c3a73c480d38aa038/test_suite/tests/compile-fail/unknown-attribute/container.rs#L5) is testing [this compile-time failure](https://github.com/serde-rs/serde/blob/c488cec6412202280b99748c3a73c480d38aa038/serde_codegen_internals/src/attr.rs#L274-L275). I would like a similar test for any `cx.error` failures added in this PR. That way we know we don't regress any of these error messages if serde_codegen_internals ever needs to be refactored. An example test would look like `#[serde(into = ")")] //~^ failed to parse type: ")"` (or whatever the message is).